prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt

```diff
@@ -440,22 +452,29 @@ class FHIRRouter(
     /**
      * Takes a [bundle] and [filter] and optionally a [focusResource], evaluates if the bundle passes the filter, or the
      * opposite if [reverseFilter] is true. If the filter is null or empty, return [defaultResponse].
-     * @return Boolean indicating if the bundle passes the filter or not
-     *         String to use when logging the filter result
+     * @param filter Filter that will be evaluated
+     * @param bundle FHIR Bundle that will be evaluated
+     * @param defaultResponse Response returned if the filter is null or empty
+     * @param reverseFilter Optional flag used to reverse the result of the filter evaluation
+     * @param focusResource Starting point for the evaluation, can be [bundle] if checking from root
+     * @param singlePass Optional flag used to allow elements that pass any of the filter expressions
+     * @return Pair: Boolean indicating if the bundle passes the filter or not
+     *         and String to use when logging the filter result
      */
     internal fun evaluateFilterCondition(
         filter: ReportStreamFilter?,
         bundle: Bundle,
         defaultResponse: Boolean,
         reverseFilter: Boolean = false,
         focusResource: Base = bundle,
+        singlePass: Boolean = false
     ): Pair<Boolean, String?> {
         // the filter needs to check all expressions passed in, or if the filter is null or empty it will return the
         // default response
         if (filter.isNullOrEmpty()) {
             return Pair(defaultResponse, "defaultResponse")
         }
-
+        val successFilters = mutableListOf<String>()
```

Comment:
It seems like we're using this list as more of a boolean, it'll either be empty or it won't, but we're never looking at it's contents beyond that. I'd suggest replacing this with a boolean `passedOne` initialized to `false`, and later where you add to this list, instead set the boolean to `true`. Then when evaluating the overall result, just check for `singlePass && passedOne`.